bluepyopt/ephys/create_acc.py

```diff
@@ -0,0 +1,366 @@
+       _nrn2arb_var[param.name].conv is not None:
+        return format_float(_nrn2arb_var[param.name].conv(float(param.value)))
+    else:
+        return param.value
+
+
+def _arb_is_global_param(loc, param):
+    """Returns if location-specific variable is a global one in Arbor."""
+    return loc == 'all' and param.name in ['membrane-capacitance']
+
+
+# Define BluePyOpt to Arbor region mapping
+# (relabeling locations to SWC convention)
+# Remarks:
+#  - using SWC convetion: 'dend' for basal dendrite, 'apic' for apical dendrite
+#  - myelinated is unsupported in Arbor
+ArbRegion = namedtuple('ArbRegion', 'ref, defn')
+
+
+def _make_region(region, expr=None):
+    """Create Arbor region with region name and defining expression
+    (name for decor, defined in label_dict) or region expression only
+    (for decor, no defined label in label_dict)."""
+    if expr is not None:
+        return ArbRegion(ref='(region \"%s\")' % region,
+                         defn='(region-def \"%s\" %s)' % (region, expr))
+    else:
+        return ArbRegion(ref=region, defn=expr)
+
+
+def _make_tagged_region(region, tag):
+    return _make_region(region, '(tag %i)' % tag)
+
+
+_loc2arb_region = dict(
+    # defining "all" region for convenience here, else use
+    # all=_arb_defined_region('(all)') to omit "all" in label_dict
+    all=_make_region('all', '(all)'),
+    somatic=_make_tagged_region('soma', 1),
+    axonal=_make_tagged_region('axon', 2),
+    basal=_make_tagged_region('dend', 3),
+    apical=_make_tagged_region('apic', 4),
+    myelinated=_make_region(None),
+)
+
+# # Generated with NMODL in arbor/mechanisms
+# import os, re, pprint
+
+# nmodl_pattern = '^\s*%s\s+((?:\w+\,\s*)*?\w+)\s*?$'
+# suffix_pattern = nmodl_pattern % 'SUFFIX'
+# globals_pattern = nmodl_pattern % 'GLOBAL'
+# ranges_pattern = nmodl_pattern % 'RANGE'
+
+# def process_nmodl(nmodl_str):
+#     nrn = re.search(r'NEURON\s+{([^}]+)}', nmodl_str,
+#                     flags=re.MULTILINE).group(1)
+#     suffix = re.search(suffix_pattern, nrn,
+#                        flags=re.MULTILINE)
+#     suffix = suffix if suffix is None else suffix.group(1)
+#     globals = re.search(globals_pattern, nrn,
+#                         flags=re.MULTILINE)
+#     globals = globals if globals is None \
+#               else re.findall(r'\w+', globals.group(1))
+#     ranges = re.search(ranges_pattern, nrn,
+#                        flags=re.MULTILINE)
+#     ranges = ranges if ranges is None \
+#              else re.findall(r'\w+', ranges.group(1))
+#     return dict(globals=globals, ranges=ranges)  # suffix skipped
+
+# mechs = dict()
+# for cat in ['allen', 'bbp', 'default']:
+#     mechs[cat] = dict()
+#     cat_dir = 'arbor/mechanisms/' + cat
+#     for f in os.listdir(cat_dir):
+#         with open(os.path.join(cat_dir,f)) as fd:
+#             print(f"Processing {f}", flush=True)
+#             mechs[cat][f[:-4]] = process_nmodl(fd.read())
+# pprint.pprint(mechs)
+
+
+_arb_mechs = dict(
```

Comment:
This can be put in a static file (JSON), to be accessed only from within the `_arb_nmodl_global_translate` function instead of being global.

Reply:
or in multiple files, one for each mechanism category.

Reply:
Do you mean `_arb_mechs` as a function attribute of `_arb_nmodl_global_translate`? The function operates on individual mechanisms and I would prefer not to re-read the JSON into a local variable every time it is invoked (the data should be cached between function calls). It could be read into a local variable in the invoking function, `_arb_nmodl_global_translate_local`, though. Where should I put the JSON in the repo?

Reply:
> It could be read into a local variable in the invoking function, _arb_nmodl_global_translate_local, though

Yes, exactly. That way you won't have to read it multiple inside the loop and it won't be a global dict anymore (i.e. other functions cannot access or modify it accidentally).

I think it can be put in a `static` folder under ephys `ephys/static/arbor_mechanisms.json` to denote that it's not code but a static file. What do you say @wvangeit @DrTaDa @AurelienJaquier ?

Reply:
It sounds good to me.

Reply:
Done in f2ca1ed.